### PR TITLE
IF-4277 Add workflow config

### PR DESCRIPTION
**Pull request by ghost**
This change allow us to use GitHub Actions enabling CI/CD.

### CODE REVIEW

**Review by ManabuMiwa — APPROVED** [review]
LGTM as far as I understand. :)

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+          restore-keys: |
+            ${{ runner.os }}-go-
+
+      - name: Download modules
+        if: steps.cache.outputs.cache-hit != 'true'
+        run: go mod download
+
+      # TODO: Return non-zero exit code
+      - name: Format
+        run: gofmt -d -e -l .
```
Comment (kazyshr) [inline .github/workflows/test.yml]: Can you call tasks in Makefile for maintenance and consistency? Please add and update them if needed.

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+      - name: Cache modules
+        uses: actions/cache@v1
+        with:
+          path: ~/go/pkg/mod
+          key: ${{ runner.os }}-go-${{ hashFiles('**/go.sum') }}
+          restore-keys: |
+            ${{ runner.os }}-go-
+
+      - name: Download modules
```
Comment (kazyshr) [inline .github/workflows/test.yml]: I guess this step is not necessary because modules get download by go test command.

**Review by kazyshr — CHANGES_REQUESTED** [review]
Excellent!! I commented minor requests.

File: .github/workflows/test.yml (hunk 1 above)
Comment (ghost) [reply .github/workflows/test.yml]: Done

File: .github/workflows/test.yml (hunk 2 above)
Comment (ghost) [reply .github/workflows/test.yml]: My understanding is that `-i` option is necessary if you want to download dependencies:

```
go test -i
```

, and test itself is not executed with this option.

But if what you say is true, the config can be more simple.
Could you please give me some documents describing it.

File: Makefile [hunk 3]
```diff
@@ -2,9 +2,12 @@ fmt:
 	gofmt -s -w .
 
 fmtcheck:
-	(! gofmt -s -d . | grep '^')
```
Comment (ghost) [inline Makefile:5]: Current version of `gofmt` outputs stderr instead of stdout if there's some failure.

File: testing/util_test.go [hunk 4]
```diff
@@ -69,22 +68,22 @@ func TestNormalizeURL(t *testing.T) {
 }
 
 func TestNormalizePathURL(t *testing.T) {
-	baseDir, _ := os.Getwd()
+	baseDir := "/test/path"
```
Comment (ghost) [inline testing/util_test.go:71]: Probably it's not necessary to use actual file path:)

File: .github/workflows/test.yml (hunk 2 above)
Comment (kazyshr) [reply .github/workflows/test.yml]: > Standard commands like go build or go test will automatically add new dependencies as needed to satisfy imports (updating go.mod and downloading the new dependencies).

https://github.com/golang/go/wiki/Modules#daily-workflow

File: Makefile (hunk 3 above)
Comment (kazyshr) [reply Makefile:5]: I want to get non-zero exit status when format check fails. That's why I wrote the task like this.

Comment (ghost) [reply Makefile:5]: ahh, that's my fault, I was running gofmt to files which has invalid syntax. (In this case gofmt returns non-zero exit status)
I've rollbacked it:)

**Review by kazyshr — APPROVED** [review]
Great!!

**Pull request merged**